fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/codec/LuceneOptimizedBlendedInfixSuggesterWithoutTermVectors.java

```diff
@@ -0,0 +1,291 @@
+        ft.setIndexOptions(indexOptions);
+        ft.setOmitNorms(true);
+
+        return ft;
+    }
+
+    @Override
+    protected List<LookupResult> createResults(IndexSearcher searcher, TopFieldDocs hits, int num, CharSequence key,
+                                               boolean doHighlight, Set<String> matchedTokens, String prefixToken)
+            throws IOException {
+
+        NavigableSet<LookupResult> results = new TreeSet<>(LOOKUP_COMP);
+
+        // we reduce the num to the one initially requested
+        int actualNum = num / numFactor;
+
+        for (int i = 0; i < hits.scoreDocs.length; i++) {
+            FieldDoc fd = (FieldDoc) hits.scoreDocs[i];
+
+            BinaryDocValues textDV = MultiDocValues.getBinaryValues(searcher.getIndexReader(), TEXT_FIELD_NAME);
+            assert textDV != null;
+
+            textDV.advance(fd.doc);
+
+            final String text = textDV.binaryValue().utf8ToString();
+            long weight = (Long) fd.fields[0];
+
+            // This will just be null if app didn't pass payloads to build():
+            // TODO: maybe just stored fields?  they compress...
+            BinaryDocValues payloadsDV = MultiDocValues.getBinaryValues(searcher.getIndexReader(), "payloads");
+
+            BytesRef payload;
+            if (payloadsDV != null) {
+                if (payloadsDV.advance(fd.doc) == fd.doc) {
+                    payload = BytesRef.deepCopyOf(payloadsDV.binaryValue());
+                } else {
+                    payload = new BytesRef(BytesRef.EMPTY_BYTES);
+                }
+            } else {
+                payload = null;
+            }
+
+            double coefficient;
+            if (text.startsWith(key.toString())) {
+                // if hit starts with the key, we don't change the score
+                coefficient = 1;
+            } else {
+                coefficient = createCoefficient(matchedTokens, prefixToken, text, payload);
+            }
+            if (weight == 0) {
+                weight = 1;
+            }
+            if (weight < 1 / LINEAR_COEF && weight > -1 / LINEAR_COEF) {
+                weight *= 1 / LINEAR_COEF;
+            }
+            long score = (long) (weight * coefficient);
+
+            LookupResult result;
+            if (doHighlight) {
+                result = new LookupResult(text, highlight(text, matchedTokens, prefixToken), score, payload);
+            } else {
+                result = new LookupResult(text, score, payload);
+            }
+            
+            if (results.size() >= actualNum) {
+                if (results.first().value < result.value) {
+                    results.pollFirst();
+                } else {
+                    continue;
+                }
+            }
+            results.add(result);
+        }
+
+        return new ArrayList<>(results.descendingSet());
+    }
+
+    private double createCoefficient(Set<String> matchedTokens, String prefixToken,
```

Comment:
I think it would be helpful for this method to have a link to the upstream `BlendedInfixSuggester` version. If I am not mistaken, up to formatting conventions, this class is identical to that one, _except_ for the one method. But due to the way it is implemented, class inheritance is, unfortunately, not possible. So we have a big copy and then this change.

Reply:
Added javadoc